libavcodec/qsvdec.c

```diff
@@ -344,6 +349,23 @@ static int qsv_decode_header(AVCodecContext *avctx, QSVContext *q,
         return ff_qsv_print_error(avctx, ret,
                 "Error decoding stream header");
 
+    if (avctx->color_range == AVCOL_RANGE_UNSPECIFIED)
+        avctx->color_range = video_signal_info.VideoFullRange ? AVCOL_RANGE_JPEG : AVCOL_RANGE_MPEG;
+
+    if (video_signal_info.ColourDescriptionPresent) {
+        if (avctx->color_primaries == AVCOL_PRI_UNSPECIFIED)
```

Comment:
Will avctx->color_primaries mismatch with video_signal_info.ColourPrimaries, if this is false?

Reply:
User should fill the color properties after creating the codec context if  this is false.

Reply:
According to https://github.com/FFmpeg/FFmpeg/blob/master/libavcodec/avcodec.h#L942-L968, these property should be set by libavcodec for decoding, I changed the code to set these properties when video_signal_info.ColourDescriptionPresent is 1.